### ISSUE TITLE
No archive modification by DND

### ISSUE THREAD

**Issue by tenplus1** [lxqt/lxqt-archiver#230]
It seems that I am unable to add a file to an archive by dragging & dropping a file into the lxqt-archiver window/folder, file-roller allows you to do this using PcManFm-Qt although since this is based on the Engrampa it may be missing.

**Comment by tsujan** [comment]
No, DND into archives isn't implemented and I don't think it's a good idea because archives might be modified by mistake.

Use "Add File" and "Add Folder" actions.

> since this is based on the Engrampa it may be missing

The core code is based on Engrampa — actually, on file-roller — but the GUI isn't.

**Comment by tsujan** [comment]
I changed the title because DND is reserved for opening archives.

**Comment by tenplus1** [comment]
Thanks for the quick reply, unfortunately DND is something I use regularly for easy of use and simplicity so lack thereof only means returning to file-roller and not this good qt implementation.l

**Comment by tsujan** [comment]
@tenplus1 
The feature you want could be implemented easily but it would clash with the current behavior of lxqt-archiver. Maybe, it can be implemented as an option, with the current behavior as the default. If so, we might need to add a Preferences dialog to lxqt-archiver or....

**Comment by Charadon** [comment]
Hi, I'm also wishing that DND was supported. An option at the very least would be very nice, since every other archiver has it

**Comment by FinGu** [comment]
This also gets an up from me

**Comment by FinGu** [comment]
a solution i can think of would be to open a new archive here:

<img width="528" height="1724" alt="Image" src="https://github.com/user-attachments/assets/539a92d7-66af-40d4-99d6-13e6a0ed127d" />

and add to the current file here:

<img width="2488" height="1644" alt="Image" src="https://github.com/user-attachments/assets/c11e6153-04b7-4436-867b-f51126d25800" />

**Comment by tsujan** [comment]
I agree with @isf63.

And now, after 4 years, I don't see this feature as a good idea. What other apps may do is irrelevant.

The user can already add files/folders when supported. Duplicating this feature to make it "easier" by adding a Preferences dialog whose existence wouldn't be justified otherwise is a bad idea because, duplication aside, it's about changing archives, while archives may be made to preserve files and protecting them from accidental changes. And if a prompt is needed — which *is* needed — it won't become "easier".
